profiler/profiler.go

```diff
@@ -211,6 +212,12 @@ func (p *profiler) run() {
 // an item.
 func (p *profiler) collect(ticker <-chan time.Time) {
 	defer close(p.out)
+	var (
+		// mu guards completed
+		mu        sync.Mutex
+		completed []*profile
```

Comment:
NIT: mu might not be needed if pre-allocating `completed` to the right size and putting each profile in its own slot when its done.

Reply:
I think I'll keep it this way for now for this reason: What if one of the profiles fails? Do I put `nil` at the index where the profile would go and check for nil before adding it to the batch at the end? Also if we go back to a state where some profile types produce multiple profiles (like delta + non-delta) then we'd need to do more bookkeeping to figure out the offset in the array where the profiles should go for each type.